blueflood-core/src/main/java/com/rackspacecloud/blueflood/threading/ThreadPoolSizedExecutorServiceAdapter.java

```diff
@@ -0,0 +1,104 @@
+package com.rackspacecloud.blueflood.threading;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.*;
+
+/**
+ * An adapter class so that we can use {@code ThreadPoolExecutor} with
+ * {@link SizedExecutorService}, to support dependency injection in the tests.
+ *
+ * TODO:
+ * {@link com.rackspacecloud.blueflood.service.RollupBatchWriter#enqueueRollupForWrite}
+ * makes calls to {@link #getActiveCount()} and {@link #getPoolSize()}, which
+ * means it must be hard-coded to use the {@link ThreadPoolExecutor} class
+ * instead of the {@link ExecutorService} interface. Once that method is fixed
+ * and no longer relies on those calls, it can be changed to use just
+ * {@link ExecutorService}, and both
+ * {@link ThreadPoolSizedExecutorServiceAdapter} and
+ * {@link SizedExecutorService} should be removed.
+ */
+public class ThreadPoolSizedExecutorServiceAdapter implements SizedExecutorService {
```

Comment:
I dont see a compelling reason to create this class. I have a feeling we can get by without this class. I think leaving the original signature of LocatorFetchRunnable to have ThreadPoolExecutor as it was having before, is not such a bad thing if we can avoid creating this class.